bigtable-grpc-interface/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java

```diff
@@ -210,53 +231,46 @@ public BigtableSession(BigtableOptions options, ExecutorService batchPool, Event
     this.batchPool = batchPool;
     this.options = options;
     LOG.info("Opening connection for project %s on data host %s, "
-        + "table admin host %s, using transport %s.", options.getProjectId(),
-      options.getDataHost(), options.getTableAdminHost(),
-      TransportOptions.BigtableTransports.HTTP2_NETTY_TLS);
+        + "table admin host %s.", options.getProjectId(),
+      options.getDataHost(), options.getTableAdminHost());
     this.scheduledRetries = scheduledRetries;
   }
 
-  protected TransportOptions createTransportOptions(InetAddress host) {
-    return new TransportOptions(
-        TransportOptions.BigtableTransports.HTTP2_NETTY_TLS,
-        host,
-        options.getPort(),
-        TransportOptions.SSL_CONTEXT_FACTORY,
-        elg);
-  }
-
-  public synchronized BigtableClient getDataClient() {
+  public synchronized BigtableClient getDataClient() throws IOException {
     if (this.client == null) {
-      Channel channel = createChannel(
-        createTransportOptions(options.getDataHost()),
-        options.getChannelCount());
+      Channel channel = createChannel(options.getDataHost(), options.getChannelCount());
       this.client = new BigtableGrpcClient(channel, batchPool, options.getRetryOptions());
     }
-
     return client;
   }
 
-  public synchronized BigtableTableAdminClient getTableAdminClient() {
+  public synchronized BigtableTableAdminClient getTableAdminClient() throws IOException {
     if (this.tableAdminClient == null) {
-      Channel channel = createChannel(
-        createTransportOptions(options.getTableAdminHost()),
-        1);
+      Channel channel = createChannel(options.getTableAdminHost(), 1);
       this.tableAdminClient = new BigtableTableAdminGrpcClient(channel);
     }
     return tableAdminClient;
   }
 
-  public synchronized BigtableClusterAdminClient getClusterAdminClient() {
+  public synchronized BigtableClusterAdminClient getClusterAdminClient() throws IOException {
     if (this.clusterAdminClient == null) {
-      Channel channel = createChannel(
-        createTransportOptions(options.getClusterAdminHost()),
-        1);
+      Channel channel = createChannel(options.getClusterAdminHost(), 1);
       this.clusterAdminClient = new BigtableClusterAdminGrpcClient(channel);
     }
 
     return clusterAdminClient;
   }
 
+  private InetAddress getHost(String hostName) throws IOException {
```

Comment:
You could already create here the InetSocketAddess here, and avoid passing InetAddess to later add the port.